crates/oxc_transformer/src/es2022/class_properties/static_prop_init.rs

```diff
@@ -221,6 +221,7 @@ impl<'a, 'ctx, 'v> VisitMut<'a> for StaticInitializerVisitor<'a, 'ctx, 'v> {
             // `object.#prop = value`, `object.#prop += value`, `object.#prop ??= value` etc
             Expression::AssignmentExpression(_) => {
                 self.class_properties.transform_assignment_expression(expr, self.ctx);
+                self.class_properties.transform_super_assignment_expression(expr, self.ctx);
```

Comment:
I think this is the cause of the panic in conformance. Problem is that `transform_assignment_expression` sometimes mutates `expr` so it's not an `AssignmentExpression` any more. So then `transform_super_assignment_expression` enters `unreachable!()`.

So we need to check that it's still an `AssignmentExpression`:

```suggestion
                if matches!(expr, Expression::AssignmentExpression(_)) {
                    self.class_properties.transform_super_assignment_expression(expr, self.ctx);
                }
```

Having to do this check again is inefficient, but don't worry - this visitor will be removed anyway in my forthcoming refactoring, so we can change it then.